src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/ValidateResourceIdFilterAttribute.cs

```diff
@@ -24,19 +25,20 @@ public override void OnActionExecuting(ActionExecutingContext context)
             if (context.RouteData.Values.TryGetValue(KnownActionParameterNames.Id, out var actionId) &&
                 context.ActionArguments.TryGetValue(KnownActionParameterNames.Resource, out var parsedModel))
             {
-                if (!string.Equals((string)actionId, ((Resource)parsedModel).Id, StringComparison.Ordinal))
```

Comment:
Moves up Id null check if-block since it is not reachable as this usecase getting handled by id not matched if-block